init/modules/EESSI/2023.06.lua

```diff
@@ -17,56 +17,123 @@ local eessi_os_type = "linux"
 local eessi_eprefix = pathJoin(eessi_prefix, "compat", eessi_os_type, eessi_cpu_family)
 local eessi_software_path = pathJoin(eessi_prefix, "software", eessi_os_type, eessi_software_subdir)
-local eessi_module_path = pathJoin(eessi_software_path, "modules", "all")
+local eessi_modules_subdir = pathJoin("modules", "all")
+local eessi_module_path = pathJoin(eessi_software_path, eessi_modules_subdir)
 local eessi_site_software_path = string.gsub(eessi_software_path, "versions", "host_injections")
-local eessi_site_module_path = pathJoin(eessi_site_software_path, "modules", "all")
+local eessi_site_module_path = pathJoin(eessi_site_software_path, eessi_modules_subdir)
 setenv("EPREFIX",  eessi_eprefix)
+eessiDebug("Setting EPREFIX to " .. eessi_eprefix)
 setenv("EESSI_CPU_FAMILY", eessi_cpu_family)
+eessiDebug("Setting EESSI_CPU_FAMILY to " .. eessi_cpu_family)
 setenv("EESSI_SITE_SOFTWARE_PATH", eessi_site_software_path)
+eessiDebug("Setting EESSI_SITE_SOFTWARE_PATH to " .. eessi_site_software_path)
 setenv("EESSI_SITE_MODULEPATH", eessi_site_module_path)
+eessiDebug("Setting EESSI_SITE_MODULEPATH to " .. eessi_site_module_path)
 setenv("EESSI_SOFTWARE_SUBDIR", eessi_software_subdir)
+eessiDebug("Setting EESSI_SOFTWARE_SUBDIR to " .. eessi_software_subdir)
 setenv("EESSI_PREFIX", eessi_prefix)
+eessiDebug("Setting EESSI_PREFIX to " .. eessi_prefix)
 setenv("EESSI_EPREFIX", eessi_eprefix)
+eessiDebug("Setting EPREFIX to " .. eessi_eprefix)
 prepend_path("PATH", pathJoin(eessi_eprefix, "bin"))
-prepend_path("PATH", pathJoin(eessi_eprefix, "usr/bin"))
+eessiDebug("Adding " .. pathJoin(eessi_eprefix, "bin") .. " to PATH")
+prepend_path("PATH", pathJoin(eessi_eprefix, "usr", "bin"))
+eessiDebug("Adding " .. pathJoin(eessi_eprefix, "usr", "bin") .. " to PATH")
 setenv("EESSI_SOFTWARE_PATH", eessi_software_path)
+eessiDebug("Setting EESSI_SOFTWARE_PATH to " .. eessi_software_path)
 setenv("EESSI_MODULEPATH", eessi_module_path)
+eessiDebug("Setting EESSI_MODULEPATH to " .. eessi_module_path)
 if ( mode() ~= "spider" ) then
     prepend_path("MODULEPATH", eessi_module_path)
+    eessiDebug("Adding " .. eessi_module_path .. " to MODULEPATH")
 end
-prepend_path("LMOD_RC", pathJoin(eessi_software_path, "/.lmod/lmodrc.lua"))
-prepend_path("MODULEPATH", eessi_site_module_path)
+prepend_path("LMOD_RC", pathJoin(eessi_software_path, ".lmod", "lmodrc.lua"))
+eessiDebug("Adding " .. pathJoin(eessi_software_path, ".lmod", "lmodrc.lua") .. " to LMOD_RC")
 setenv("LMOD_PACKAGE_PATH", pathJoin(eessi_software_path, ".lmod"))
+eessiDebug("Setting LMOD_PACKAGE_PATH to " .. pathJoin(eessi_software_path, ".lmod"))
+
+-- the accelerator may have an empty value and we need to give some flexibility
+-- * construct the path we expect to find
+-- * then check it exists
+-- * then update the modulepath
+if not (archdetect_accel == nil or archdetect_accel == '') then
+    eessi_accel_software_subdir = os.getenv("EESSI_ACCEL_SOFTWARE_SUBDIR_OVERRIDE") or eessi_software_subdir
```

Comment:
```suggestion
    -- The CPU subdirectory of the accelerator installations is _usually_ the same as host CPU, but this can be overridden
    eessi_accel_software_subdir = os.getenv("EESSI_ACCEL_SOFTWARE_SUBDIR_OVERRIDE") or eessi_software_subdir
```